### PR TITLE
couple of fixes

### PR DESCRIPTION
**Pull request by hugohabicht01**
- get rid of bullshit that made the website horrible to interact with
- replace almost all occurences of `role` strings with an 'enum' (due to the recent switch to sqlite)
- adds slides pages (to replace the presentations domain that has terrible UX)

### CODE REVIEW

**Review by 456dev — CHANGES_REQUESTED** [review]
i might just have a different idea about what is needed, but the ability to link / view 1 "slides" object individually would be useful for linking from the events pages etc.
also remove the bun lockfile for modern yarn

File: bun.lockb
[the remark this answers is not in the document]
Comment (hugohabicht01) [reply bun.lockb:1]: good shout, that file shouldn't have made its way into the commits anyway...

File: prisma/schema.prisma [hunk 1]
```diff
@@ -40,6 +41,20 @@ model Event {
...
 
+model Slides {
+  id        String   @id @default(cuid())
+  link      String
+  name      String
+  speaker   String
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+  creator   User     @relation(fields: [creatorId], references: [id])
+
```
[the remark this answers is not in the document]
Comment (hugohabicht01) [reply prisma/schema.prisma:53]: that gets done by the mimetype field, how we should render it, should mainly depend on the mimetype

File: pages/slides.vue
[the remark this answers is not in the document]
Comment (hugohabicht01) [reply pages/slides.vue:1]: the links can all be accessed directly

**Review by 456dev — APPROVED** [review]
LGTM

**Pull request merged**